### PR TITLE
conman 0.3.1 (new formula)

### PR DESCRIPTION
**Pull request by jchonig**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----
Adding new formula for [conman](https://github.com/dun/conman), a remote console manager and client.

### CODE REVIEW

**Comment by jchonig** [comment]
I disabled arm64 builds then realized that the failure reason seems to be in the test infrastructure, not the formula or the code:

```
Error: Value cannot be null. (Parameter 'ContainerId')
```

**Comment by jchonig** [comment]
Fixed my stupid mistake of committing conman.rb in the wrong directory.

File: Formula/conman.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  end
+
+  service do
+    run [opt_sbin/"conmand", "-F", "-c", etc/"conman.conf"]
+    keep_alive true
+  end
+
+  test do
+    system bin/"conman", "-V"
+    system sbin/"conmand", "-V"
```
Comment (chenrui333) [inline Formula/conman.rb:34]: Can you propose a more substantial test than checking the version or calling for help? These tests rarely ever fail even for broken installations.

>We want tests that don’t require any user input and test the basic functionality of the application. For example `foo build-foo input.foo` is a good test and (despite their widespread use) `foo --version` and `foo --help` are bad tests.

Ref. https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula

File: Formula/conman.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+                          "--disable-silent-rules",
+                          "--prefix=#{prefix}"
+    system "make", "install"
+  end
+
+  def caveats
+    <<~EOS
+      Before starting the conmand service, configure some consoles in #{etc}/conman.conf.
+    EOS
+  end
```
Comment (chenrui333) [inline Formula/conman.rb:25]: If this is in the upstream, we probably dont need this in homebrew-core, if not, we can contribute back to the upstream? Might be good to use a general sample `conman.conf` if exists.

File: Formula/conman.rb [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  depends_on "autoconf" => :build
+  depends_on "automake" => :build
+  depends_on "freeipmi"
+
+  def install
+    system "./bootstrap"
+    system "./configure", "--disable-debug",
+                          "--disable-dependency-tracking",
+                          "--disable-silent-rules",
+                          "--prefix=#{prefix}"
```
Comment (chenrui333) [inline Formula/conman.rb:17]:
```suggestion
    system "./configure", *std_configure_args, "--disable-silent-rules"
```

File: Formula/conman.rb [hunk 4]
```diff
@@ -0,0 +1,36 @@
+class Conman < Formula
+  desc "Serial console management program supporting a large number of devices"
+  homepage "https://github.com/dun/conman"
+  url "https://github.com/dun/conman/archive/conman-0.3.1.tar.gz"
+  sha256 "cd47d3d9a72579b470dd73d85cd3fec606fa5659c728ff3c1c57e970f4da72a2"
+  license "GPL-3.0-only"
```
Comment (chenrui333) [inline Formula/conman.rb:6]: https://github.com/dun/conman/blob/master/server.c#L1-L22
```suggestion
  license "GPL-3.0-or-later"
```

File: Formula/conman.rb (hunk 2 above)
Comment (jchonig) [reply Formula/conman.rb:26]: There is a sample configuration, but it does not define any consoles.  You need to have a console to connect to and there really isn't a default.

This package is useful w/o configuring conmand, the conman client can be configured to talk to a remote server.

File: Formula/conman.rb (hunk 1 above)
Comment (jchonig) [reply Formula/conman.rb]: Tests added

File: Formula/conman.rb [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+  test do
+    assert_match "conman-#{version}", shell_output("#{bin}/conman -V 2>&1")
+    assert_match "conman-#{version} FREEIPMI", shell_output("#{sbin}/conmand -V 2>&1")
+
+    conffile = testpath/"conman.conf"
+    conffile.write(<<~EOS,
+      console name="test-sleep1" dev="/bin/sleep 30"
+      console name="test-sleep2" dev="/bin/sleep 30"
+    EOS
+                  )
```
Comment (chenrui333) [inline Formula/conman.rb:39]:
```suggestion
    conffile.write <<~EOS
      console name="test-sleep1" dev="/bin/sleep 30"
      console name="test-sleep2" dev="/bin/sleep 30"
    EOS
```

File: Formula/conman.rb (hunk 2 above)
Comment (chenrui333) [reply Formula/conman.rb:23]: 👍

**Review by chenrui333 — DISMISSED** [review]
LGTM

File: Formula/conman.rb [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+  sha256 "cd47d3d9a72579b470dd73d85cd3fec606fa5659c728ff3c1c57e970f4da72a2"
+  license "GPL-3.0-or-later"
+
+  depends_on "autoconf" => :build
+  depends_on "automake" => :build
+  depends_on "freeipmi"
+
+  def install
+    system "./bootstrap"
+    system "./configure", *std_configure_args, "--disable-silent-rules"
```
Comment (carlocab) [inline Formula/conman.rb:14]:
```suggestion
    system "./configure", *std_configure_args, "--disable-silent-rules", "--sysconfdir=#{etc}"
```
Let's not install configuration files into the cellar. These will get blown away by `brew upgrade`.

File: Formula/conman.rb [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+    system "make", "install"
+    inreplace Dir["#{pkgshare}/examples/*.exp"], "/usr/share/", "#{share}/"
```
Comment (carlocab) [inline Formula/conman.rb:16]:
```suggestion
    inreplace pkgshare.glob("examples/*.exp"), "/usr/share/", "#{opt_share}/"
```

**Review by carlocab — APPROVED** [review]

**Pull request closed without merging**